packages/idempotency/src/idempotencyDecorator.ts

```diff
@@ -0,0 +1,69 @@
+import { AnyFunction, ItempotentFunctionOptions } from './types';
+import { makeIdempotent } from './makeIdempotent';
+
+/**
+ * Use this decorator to make your lambda handler itempotent.
+ * You need to provide a peristance layer to store the idempotency information.
+ * At the moment we only support `DynamodbPersistenceLayer`.
+ *
+ * @example
+ * ```ts
+ * import {
+ *   DynamoDBPersistenceLayer,
+ *   idempotentLambdaHandler
+ * } from '@aws-lambda-powertools/idempotency'
+ *
+ * class MyLambdaFunction {
+ *   @idempotent({ persistenceStore: new DynamoDBPersistenceLayer() })
+ *   async handler(event: any, context: any) {
+ *     return "Hello World";
+ *   }
+ * }
+ * export myLambdaHandler new MyLambdaFunction();
+ * export const handler = myLambdaHandler.handler.bind(myLambdaHandler);
+ * ```
+ *
+ * Similar to decoratoring a handler you can use the decorator on any other function.
+ * @example
+ * ```ts
+ * import {
+ *  DynamoDBPersistenceLayer,
+ *  idempotentFunction
+ *  } from '@aws-lambda-powertools/idempotency'
+ *
+ *  class MyClass {
```

Comment:
I think both snippets are missing the `implements LambdaInterface` and might be missing some closing braces.